.github/workflows/deploy_docs.yml

```diff
@@ -0,0 +1,45 @@
+
+name: deploy docs
+on:
+  release:
+    types: [published]
+    
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: install prerequisites
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y \
+          libdrm-dev \
+          libegl1-mesa-dev \
+          libgl1-mesa-dev \
+          libx11-dev \
+          libxext-dev \
+          libxfixes-dev \
+          libwayland-dev \
+          libx11-xcb-dev \
+           libxcb-dri3-0 \
+          doxygen 
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.10' 
+
+    - name: Build
+      run: |
+        python -m pip install ninja
+        python -m pip install meson
+        meson _build -Denable_docs=true
+        meson compile -C _build
+
+    - name: Build echo
+      run:  echo ${{ github.ref }}
+
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v3
```

Comment:
I did not look closely into github pages, but there seems to be standard ways of doing that. Why use custom github action?

* https://github.com/actions/deploy-pages
* https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site
* https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site

Reply:
I tried this, but I can't deploy with release tag.